server/src/test/java/org/opensearch/index/store/remote/filecache/FullFileCachedIndexInputTests.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store.remote.filecache;
+
+import org.apache.lucene.store.IndexInput;
+
+import java.io.IOException;
+
+public class FullFileCachedIndexInputTests extends FileCachedIndexInputTests {
+    private FullFileCachedIndexInput fullFileCachedIndexInput;
+
+    @Override
+    protected void setupIndexInputAndAddToFileCache() {
+        fullFileCachedIndexInput = new FullFileCachedIndexInput(fileCache, filePath, underlyingIndexInput);
+        fileCache.put(filePath, new FullFileCachedIndexInputImpl(fileCache, filePath, fullFileCachedIndexInput));
+    }
+
+    @Override
+    public void testClone() throws IOException {
+        setupIndexInputAndAddToFileCache();
+
+        // Since the file ia already in cache and has refCount 1, activeUsage and totalUsage will be same
+        assertTrue(isActiveAndTotalUsageSame());
+
+        // Decrementing the refCount explicitly on the file which will make it inactive (as refCount will drop to 0)
+        fileCache.decRef(filePath);
+        assertFalse(isActiveAndTotalUsageSame());
+
+        // After cloning the refCount will increase again and activeUsage and totalUsage will be same again
+        FileCachedIndexInput clonedFileCachedIndexInput1 = fullFileCachedIndexInput.clone();
+        FileCachedIndexInput clonedFileCachedIndexInput2 = clonedFileCachedIndexInput1.clone();
+        FileCachedIndexInput clonedFileCachedIndexInput3 = clonedFileCachedIndexInput2.clone();
+        assertTrue(isActiveAndTotalUsageSame());
+
+        // Closing the parent will close all the clones decreasing the refCount to 0
+        fullFileCachedIndexInput.close();
```

Comment:
You should `close` the `clonedFileCachedIndexInput1` first and verify that ref count is back to 0 and active vs total usage will not be same. This means the cache entry in FileCache i.e. `FullFileCachedIndexInputImpl` will be eligible for eviction. So now call `prune` for evicting the `FullFileCachedIndexInputImpl` entry which will call `close` on it and calling `close` on `fullFileCachedIndexInput` internally. Then verify calling `getIndexInput` on that `FullFileCachedIndexInputImpl` throws `AlreadyClosedException`

Reply:
Done in the latest revision